tools/audio/extraction/audio_extract.py

```diff
@@ -176,6 +174,135 @@ def extract_samplebank(pool : ThreadPool, extracted_dir : str, sample_banks : Li
+        fonts = sequence_font_table[font_data_offset:font_data_offset+num_fonts]
+
+        all_fonts.append(fonts)
+
+        # mark coverage for sequence font table
+        sequence_font_table_cvg[2 * i + 0] = 1
+        sequence_font_table_cvg[2 * i + 1] = 1
+        for j in range(font_data_offset-1,font_data_offset+num_fonts):
+            sequence_font_table_cvg[j] = 1
+
+        if entry.size != 0:
+            # Real sequence, queue extraction
+
+            seq_data = bytearray(entry.data(audioseq_seg, sequence_table.rom_addr))
+
+            ext = ".prg" if i in handwritten_sequences else ""
+
+            if BASEROM_DEBUG:
+                # Extract original sequence binary for comparison
+                with open(f"{extracted_dir}/baserom_audiotest/audioseq_files/seq_{i}{ext}.aseq", "wb") as outfile:
+                    outfile.write(seq_data)
+
+            extraction_xml = sequence_xmls.get(i, None)
+            if extraction_xml is None:
+                sequence_filename = f"seq_{i}"
+                sequence_name = f"Sequence_{i}"
+            else:
+                sequence_filename = extraction_xml[0]
+                sequence_name = extraction_xml[1].attrib["Name"]
+
+            # Write extraction xml entry
+            if write_xml:
+                xml = XMLWriter()
+
+                xml.write_element("Sequence", {
+                    "Name" : sequence_name,
+                    "Index" : i,
+                })
+
+                with open(f"assets/xml/audio/sequences/{sequence_filename}.xml", "w") as outfile:
+                    outfile.write(str(xml))
+
+            if i in handwritten_sequences:
+                # skip "handwritten" sequences
+                continue
+
+            disas_jobs.append((i, seq_data, sequence_name, sequence_filename, fonts))
+        else:
+            # Pointer to another sequence, checked later
+            pass
+
+    # Check full coverage
+    try:
+        if align(sequence_font_table_cvg.index(0), 16) != len(sequence_font_table_cvg):
+            # does not pad to full size, fail
+            assert False, "Sequence font table missing data"
+        # pads to full size, good
+    except ValueError:
+        pass # fully covered, good
+
+    # Check consistency of font data for the same sequence accessed via pointers
+
+    for i,entry in enumerate(sequence_table):
+        entry : AudioCodeTableEntry
+
+        # Fonts for this entry
+        fonts = all_fonts[i]
+
+        if entry.size != 0:
+            # real, ignore
+            pass
+        else:
+            # pointer, check that the fonts for this entry are the same as the fonts for the other
+            j = entry.rom_addr
+
+            fonts2 = all_fonts[j]
+
+            assert fonts == fonts2, \
+                   f"Font mismatch: Pointer {i} against Real {j}. This is a limitation of the build process."
+
+    # Disassemble to text
+
+    with ThreadPool(processes=os.cpu_count()) as pool:
```

Comment:
Samplebank extraction is definitely sped up from multiprocessing in this way. Before it would take me a minute to extract all the samples instead of ~6 seconds. I didn't see the same improves for sequences though, likely for the reasons you mentioned. I'll try to adjust it to actually work.

Reply:
I wasn't able to get it to work with `multiprocessing.Pool` since it complains about being unable to pickle `memoryview`. I've decided to just not multiprocess sequences for now, it's not that slow anyway.